usecases/auth/authentication/apikey/db_users.go

```diff
@@ -286,7 +300,12 @@ func (c *DBUser) ValidateAndExtract(key, userIdentifier string) (*models.Princip
 		return nil, fmt.Errorf("key is revoked")
 	}
 
-	c.data.Users[userId].LastUsedAt = time.Now()
+	// Last used time does not have to be exact. If we have multiple concurrent requests for the same
+	// user, only recording one of them is good enough
+	if c.data.Users[userId].TryLock() {
```

Comment:
> I think I found a valid usecase for TryLock!

So like "if someone's already updating it, I don't need to"? Nice!

Reply:
yes, if the requests are concurrent there is not much difference in time anyways!